### PR TITLE
Introduce animation edition

### PR DESCRIPTION
**Pull request by obzota**
Display the different actions with an alternative menu when clicked.
User can select the parameters via select lists and number inputs.

Add :
- Layoutview mechanism for Actionview settings
- Templating to change views in a dynamic way (toggle on/off settings)
- inheritance between the views to control the templating and minimize
  the code redundancie
- subview in MoveView to control the region extra-settings
- new set of events triggered by user actions in 'action' view. Associated
  callbacks for (re)rendering and model updates.
- attributes in actions models to match the parameters used in preview

Delete :
- unused template appear and disappear.

### CODE REVIEW

File: dahu/core/app/scripts/templates/views/workspace/actions/move.html [hunk 1]
```diff
@@ -1,7 +1,6 @@
-<div class="move">
-    {{type}}
-    {{target}}
-    {{trigger}}
-    {{finalAbs}}
-    {{finalOrd}}
-</div>
\ No newline at end of file
+<form>
+    <label for="trXchoice">Translation X</label>
```
Comment (barraq) [inline dahu/core/app/scripts/templates/views/workspace/actions/move.html]: wrong label...

File: dahu/core/app/scripts/templates/views/workspace/actions/move.html [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
+    <input type="number" value="{{trX}}" id="trXChoice_{{id}}" class="trXChoice"><br>
+    <label for="trYchoice">Translation Y</label>
```
Comment (barraq) [inline dahu/core/app/scripts/templates/views/workspace/actions/move.html]: idem

File: dahu/core/app/scripts/views/workspace/actions/move.js [hunk 3]
```diff
@@ -5,19 +5,49 @@
...
-    return Marionette.ItemView.extend({
-        template: Handlebars.default.compile(moveTemplate)
+    return ActionView.extend({
+
+        onChildviewMoveTrxChange: function () {
+            this.model.set("trX", $("#trXChoice_" + this.model.id).val());
+        },
+
+        onChildviewMoveTryChange: function () {
+            this.model.set("trY", $("#trYChoice_" + this.model.id).val());
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions/move.js]: what if the user enter "blabla" ? There is no verification that it is a number.

File: dahu/core/app/scripts/views/workspace/actions/move.js [hunk 4]
```diff
@@ -5,19 +5,49 @@
...
+        },
+
+        onRender: function () {
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions/move.js:65]: put it in onBeforeShow (see https://github.com/marionettejs/backbone.marionette/blob/master/docs/marionette.region.md#events-raised-during-show)

File: dahu/core/app/scripts/templates/views/workspace/actions/move.html (hunk 1 above)
Comment (obzota) [inline dahu/core/app/scripts/templates/views/workspace/actions/move.html]: Corrected.

File: dahu/core/app/scripts/templates/views/workspace/actions/move.html (hunk 2 above)
Comment (obzota) [inline dahu/core/app/scripts/templates/views/workspace/actions/move.html]: Corrected

File: dahu/core/app/scripts/views/workspace/actions/move.js (hunk 3 above)
Comment (obzota) [inline dahu/core/app/scripts/views/workspace/actions/move.js]: So html ' type="number" ' does nothing special for this ?

File: dahu/core/app/scripts/views/workspace/actions/move.js (hunk 4 above)
Comment (obzota) [inline dahu/core/app/scripts/views/workspace/actions/move.js:65]: Doesnt work well with 'onBeforeShow'.
Works with 'onShow'.

File: dahu/core/app/scripts/views/workspace/actions/move.js (hunk 3 above)
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions/move.js]: does the validation work within a JavaFx webview ?

Comment (obzota) [inline dahu/core/app/scripts/views/workspace/actions/move.js]: Save nothing in the model (can be 'null' or 'undefined', JSON is ' trY:"" ' )

File: dahu/core/app/scripts/views/workspace/actions/move.js [hunk 5]
```diff
@@ -5,19 +5,63 @@
...
+            var value = parseInt($(id).val());
+            if (!_.isNaN(value)){
+                this.model.set('trY',value);
+            }
+            else {
+                throw new Exceptions.IOError("Translation Y input is NaN");
+            }
+        },
+
+        onRender: function () {
```
Comment (obzota) [inline dahu/core/app/scripts/views/workspace/actions/move.js:65]: @barraq Doesnt work with 'onShow' neither with 'onBeforeShow'.
'onBeforeShow' will never display extra settings
'onShow' will not display settings on refresh (for trigger change)

**Pull request closed without merging**